scheduler/reconcile/reconcile_cluster.go

```diff
@@ -267,7 +255,7 @@ func (a *allocReconciler) computeDeploymentComplete(m allocMatrix) bool {
 	return complete
 }
 
-func (a *allocReconciler) computeDeploymentUpdates(deploymentComplete bool) {
+func (a *AllocReconciler) computeDeploymentUpdates(deploymentComplete bool) {
```

Comment:
I wouldn't mind accepting `a.Result` as a parameter and returning it here as well. I think that could turn into a huge refactoring though, so feel free to ignore. I just think if we relied less on mutating struct fields and more on method parameters it would make the code much more understandable. Perhaps we could remove result as a struct field entirely and make it purely computed by Compute!

Feel free to ignore though as the work involved might dramatically outweigh any benefits. Or at least such an effort might be better done in a followup PR.